test_plans/oneapi_discard_queue_events.asciidoc

```diff
@@ -0,0 +1,82 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl extension oneapi discard queue events
+
+This is a test plan for the extension adding
+`ext::oneapi::property::queue::discard_events` property for `sycl::queue`. The
+extension is described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_discard_queue_events.asciidoc[sycl_ext_oneapi_discard_queue_events.asciidoc].
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS` so they can
+be skipped if feature is not supported.
+
+== Tests
+
+Extension introduces new `ext::oneapi::property::queue::discard_events`
+property for `sycl::queue` and new enumerator value for the
+`sycl::info::event_command_status` enumeration. When the `queue` is created
+with the `discard_events` property the `queue` member functions returning
+events will return _invalid_ events.
+Behavior of some _invalid_ event member functions is different from the default
+event behavior and needs to be tested. The behavior when _invalid_ event is
+passed into `handler::depends_on()` function is also different and needs to be
+tested.
+
+=== Test description
```

Comment:
All the tests listed here are negative tests -- checking various error conditions.  There should be some positive tests also:

* Do all of the following twice: with a queue constructed with `discard_events` and with a queue constructed with both `discard_events` and `in_order`:
  * Submit a single_task kernel, wait on the queue, validate that the kernel ran.
  * Submit a simple parallel_for kernel, wait on the queue, validate that the kernel ran.
  * Submit an nd-range kernel, wait on the queue, validate that the kernel ran.
  * Submit a single task kernel using assert (where the assert does not trigger), wait on the queue, validate that the kernel ran.
  * Submit a single task kernel using `stream`, wait on the kernel, validate that the kernel ran.
  * Submit a single task kernel using a buffer accessor, wait on the kernel, validate that the kernel ran.

Reply:
Added.